echo-api/src/main/java/com/netflix/spinnaker/echo/api/events/Metadata.java

```diff
@@ -33,6 +36,6 @@
   private String project;
   private String application;
   private String _content_id;
-  private Map attributes;
-  private HttpHeaders requestHeaders = new HttpHeaders();
+  private Map<String, String> attributes;
+  private Map<String, List<String>> requestHeaders = new HashMap<>();
```

Comment:
@jonsie It looks like this broke code that relied on the assumption of headers being converted to lowercase by `HttpHeaders`, as in #840. I think it might be best to revert this line instead of fixing each specific downstream issue that comes up as a result, unless there was a good reason to convert this particular property to a `Map`.

Reply:
Ah sorry, I didn't catch this.  Unfortunately, we can't pull in the Spring dependency into `echo-api` as this module needs to be dependency free (basically, just Java) , which is why I removed `HttpHeaders`.

Reply:
This might do the trick:

```
private Map<String, List<String>> requestHeaders = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Reply:
https://github.com/spinnaker/echo/pull/863